### PR TITLE
Add mime.types file template and default values for it

### LINKED ISSUE #1240: Reliance on mime.types
Hey there,

excellent puppet module!

However, while experimenting with this module, I have stumbled upon a problem: the module blindly relies on the presence of a `mime.types` files (specifically in https://github.com/voxpupuli/puppet-nginx/blob/0ef72eb7af17a8255dae10ea1eb901b8894a8abf/templates/conf.d/nginx.conf.erb#L69 ). While this will work for most distributions, it does not work when using a custom config directory (`nginx::conf_dir`).

In my case, the configuration gets generated as follows (`nginx.conf`):
```
# MANAGED BY PUPPET
user www-data;
worker_processes 1;
worker_rlimit_nofile 1024;
...
http {

  include       /mnt/etc/nginx/mime.types;
  default_type  application/octet-stream;
  ...
  include /mnt/etc/nginx/conf.d/*.conf;
  include /mnt/etc/nginx/sites-enabled/*;
}
```

Since `/mnt/etc/nginx` was created by the module itself, it is initially empty and therefore lacks `mime.types`.

Upon startup, nginx then complains:
``` nginx -c /mnt/etc/nginx/nginx.conf 
nginx: [emerg] open() "/mnt/etc/nginx/mime.types" failed (2: No such file or directory) in /mnt/etc/nginx/nginx.conf:17
```

### PR DESCRIPTION
**Pull request by martialblog**
- The mine.types file can now be configured using a simple hash

#### Pull Request (PR) description
When using this module to only configure nginx and then run in a Container, we ran into an issue when we tried to store the config in a different directory. The config directory can be adjusted using the conf_dir parameter, however the module does not provide a mime.types file.

This PR fixes this issue by adding a simple mime.types.erb that can be filled with a hash, and some default values for the file.

#### This Pull Request (PR) fixes the following issues
Fixes #1240

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -147,6 +147,7 @@
   $package_source                                            = 'nginx',
   $package_flavor                                            = undef,
   $manage_repo                                               = $nginx::params::manage_repo,
+  $mime_types                                                = $nginx::params::mime_types,
```
Comment (bastelfreak) [inline manifests/init.pp]: Please add a datatype for the new parameter. Probably ```Hash[String[1], String[1]]```

File: templates/conf.d/mime.types.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# MANAGED BY PUPPET
+types {
+<% @mime_types.each do |key, value| -%>
+    <%= key %> <%= value %>;
+<% end -%>
+}
```
Comment (bastelfreak) [inline templates/conf.d/mime.types.erb]: please add a trailing newline

File: templates/conf.d/mime.types.erb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# MANAGED BY PUPPET
```
Comment (bastelfreak) [inline templates/conf.d/mime.types.erb]: Please convert this to an [epp](https://puppet.com/docs/puppet/5.5/lang_template_epp.html) template.

**Comment by martialblog** [comment]
@bastelfreak Hi, I'll take care of it

**Comment by martialblog** [comment]
I addressed the issues, but the Beaker tests blew up O_o

File: templates/conf.d/mime.types.epp [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# MANAGED BY PUPPET
+types {
+<% $nginx::mime_types.each |$key, $value| { -%>
+    <%= $key %> <%= $value %>;
+<% } -%>
```
Comment (bastelfreak) [inline templates/conf.d/mime.types.epp]: remove the `-` I guess.

File: templates/conf.d/mime.types.epp [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (bastelfreak) [inline templates/conf.d/mime.types.epp]: Please do not an empty newline.

**Comment by martialblog** [comment]
@bastelfreak Ok, thanks. I'm not too familiar with EPP Templates yet.

File: templates/conf.d/mime.types.epp (hunk 5 above)
Comment (bastelfreak) [reply templates/conf.d/mime.types.epp]: To solve a bit of the confusion I might have created. A quote from our [review docs](https://voxpupuli.org/docs/#reviewing-a-module-pr):
> * Files should always terminate with a newline if possible, with an exception being file or template fragments like those used with concat. This is the [POSIX 
standard](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206), and some tools don't handle the lack of a terminating newline properly

**Comment by martialblog** [comment]
@bastelfreak Hi, should be fine now. 

```
cat -A templates/conf.d/mime.types.epp                                                                              [11:14:57]
# MANAGED BY PUPPET$
types {$
<% $nginx::mime_types.each |$key, $value| { %>$
    <%= $key %> <%= $value %>;$
<% } %>$
}$
```

File: manifests/init.pp [hunk 6]
```diff
@@ -147,6 +147,7 @@
   $package_source                                            = 'nginx',
   $package_flavor                                            = undef,
   $manage_repo                                               = $nginx::params::manage_repo,
+  Hash[String[1], String[1]] $mime_types                     = $nginx::params::mime_types,
```
Comment (bastelfreak) [inline manifests/init.pp]: datatypes \o/

**Comment by bastelfreak** [comment]
Thanks!

**Pull request merged**